### PR TITLE
Bump minimum php version to 5.5

### PR DESCRIPTION
**Pull request by Ener-Getick**
This PR change the minimum version of php required to use FOSRestBundle.
It also updates many class paths to the new syntax sugar ``Foo::class``

### CODE REVIEW

File: DependencyInjection/Compiler/ExceptionWrapperHandlerPass.php [hunk 1]
```diff
@@ -27,7 +28,7 @@ public function process(ContainerBuilder $container)
             return;
         }
 
-        if (interface_exists('JMS\Serializer\Handler\SubscribingHandlerInterface')) {
+        if (interface_exists(SubscribingHandlerInterface::class)) {
```
Comment (xabbuh) [inline DependencyInjection/Compiler/ExceptionWrapperHandlerPass.php]: The issue with introducing changes like this right now might be that merges between branches might become trickier so we should be careful with doing them.

Comment (Ener-Getick) [inline DependencyInjection/Compiler/ExceptionWrapperHandlerPass.php]: yeah :/ we'll see if the next merges are complicated and if they are we can still revert this changes.
